packages/bundler-plugin-core/src/debug-id-upload.ts

```diff
@@ -47,34 +46,41 @@ export function debugIdUploadPlugin({
   sentryCliOptions,
   rewriteSourcesHook,
   deleteFilesAfterUpload,
-}: DebugIdUploadPluginOptions): UnpluginOptions {
-  return {
-    name: "sentry-debug-id-upload-plugin",
-    async writeBundle() {
-      let folderToCleanUp: string | undefined;
+}: DebugIdUploadPluginOptions) {
+  return async (buildArtifactPaths: string[]) => {
+    let folderToCleanUp: string | undefined;
 
-      const cliInstance = new SentryCli(null, sentryCliOptions);
+    const cliInstance = new SentryCli(null, sentryCliOptions);
 
-      try {
-        const tmpUploadFolder = await fs.promises.mkdtemp(
-          path.join(os.tmpdir(), "sentry-bundler-plugin-upload-")
-        );
+    try {
+      const tmpUploadFolder = await fs.promises.mkdtemp(
+        path.join(os.tmpdir(), "sentry-bundler-plugin-upload-")
+      );
 
-        folderToCleanUp = tmpUploadFolder;
+      folderToCleanUp = tmpUploadFolder;
 
-        const debugIdChunkFilePaths = (
-          await glob(assets, {
-            absolute: true,
-            nodir: true,
-            ignore: ignore,
-          })
-        ).filter(
-          (debugIdChunkFilePath) =>
-            debugIdChunkFilePath.endsWith(".js") ||
-            debugIdChunkFilePath.endsWith(".mjs") ||
-            debugIdChunkFilePath.endsWith(".cjs")
-        );
+      const debugIdChunkFilePaths = (
+        await glob(assets ?? buildArtifactPaths, {
+          absolute: true,
+          nodir: true,
+          ignore: ignore,
+        })
+      ).filter(
+        (debugIdChunkFilePath) =>
+          debugIdChunkFilePath.endsWith(".js") ||
+          debugIdChunkFilePath.endsWith(".mjs") ||
+          debugIdChunkFilePath.endsWith(".cjs")
+      );
 
+      if (Array.isArray(assets) && assets.length === 0) {
```

Comment:
Is `assets` always normalized to an array? otherwise, what about `assets: ""`?

Reply:
My thought process here was to just skip uploading entirely when users provide an empty array (as a means of disabling I guess). An empty string is more or less user error which we don't care to catch here. I think this will be implicitly caught by the glob pattern not finding anything and the message below then saying "check your assets option".

Reply:
Oh good point. At the time of writing I thought for some reason an empty array and an empty string the same semantic implications here but you're right, an empty string looks more like an error than an empty array.